packages/camera/lib/camera.dart

```diff
@@ -218,74 +234,88 @@ class CameraController extends ValueNotifier<CameraValue> {
     }
   }
 
-  void _applyStartStop() {
-    if (value.initialized && !_disposed) {
-      if (value.isStarted) {
-        _channel.invokeMethod(
-          'start',
-          <String, dynamic>{'textureId': _textureId},
-        );
-      } else {
-        _channel.invokeMethod(
-          'stop',
-          <String, dynamic>{'textureId': _textureId},
-        );
-      }
-    }
-  }
-
-  /// Starts the preview.
+  /// Start a video recording and save the file to [path].
   ///
-  /// If called before [initialize] it will take effect just after
-  /// initialization is done.
-  void start() {
-    value = value.copyWith(isStarted: true);
-    _applyStartStop();
+  /// A path can for example be obtained using
+  /// [path_provider](https://pub.dartlang.org/packages/path_provider).
+  ///
+  /// The file is written on the flight as the video is being recorded.
+  ///
+  /// Throws a [CameraException] if the capture fails.
+  Future<Null> startVideoRecording(String filePath) async {
+    if (!value.isInitialized || _isDisposed) {
+      throw new CameraException(
+        'Uninitialized CameraController',
+        'startVideoRecording was called on uninitialized CameraController',
+      );
+    }
+    try {
+      value = value.copyWith(isRecordingVideo: true);
+      await _channel.invokeMethod(
+        'startVideoRecording',
+        <String, dynamic>{'textureId': _textureId, 'filePath': filePath},
+      );
+    } on PlatformException catch (e) {
+      value = value.copyWith(errorDescription: e.message);
+      throw new CameraException(e.code, e.message);
+    }
   }
 
-  /// Stops the preview.
-  ///
-  /// If called before [initialize] it will take effect just after
-  /// initialization is done.
-  void stop() {
-    value = value.copyWith(isStarted: false);
-    _applyStartStop();
+  /// Stop recording.
+  Future<Null> stopVideoRecording() async {
+    if (!value.isInitialized || _isDisposed) {
+      throw new CameraException(
+        'Uninitialized CameraController',
+        'stopVideoRecording was called on uninitialized CameraController',
+      );
+    }
+    try {
+      value = value.copyWith(isRecordingVideo: false);
+      await _channel.invokeMethod(
+        'stopVideoRecording',
+        <String, dynamic>{'textureId': _textureId},
+      );
+    } on PlatformException catch (e) {
+      value = value.copyWith(errorDescription: e.message);
+      throw new CameraException(e.code, e.message);
+    }
   }
 
   /// Releases the resources of this camera.
   @override
-  Future<Null> dispose() {
-    if (_disposed) {
+  Future<Null> dispose() async {
+    if (_isDisposed) {
       return new Future<Null>.value(null);
     }
-    _disposed = true;
+    _isDisposed = true;
     super.dispose();
     if (_creatingCompleter == null) {
       return new Future<Null>.value(null);
     } else {
       return _creatingCompleter.future.then((_) async {
-        await _eventSubscription?.cancel();
```

Comment:
Yes, else you won't get error messages if "dispose" fails. I moved it here because I wasn't receiving some events at some point. I don't remember which one exactly but moving it here solved the issue.